CHANGELOG.md

```diff
@@ -233,17 +233,21 @@ This release has 74 commits from 12 contributors. Selected changes:
   formats. _format-arg_ can be `format:csv` or `format:json`. _string-arg_ can
   be a string in any format. (@aljazerzen & @snth, #1514)
 
-  ```prql
+  ```elm
```

Comment:
Should these never be `prql`?

Maybe we add a grep lint in `pre-commit` if so; otherwise folks are going to forget?

Or are they OK / actually good in `prql`?

Reply:
Thank you for your review.
The problem here is that in the book the `prql` code blocks are evaluated by the preprocessor and displayed alongside the `sql` code blocks.

If we could disable the evaluation of `prql` code blocks throughout the particular file with ignore settings, special comments or yaml front matter, I think we could continue to use `prql` code blocks there.
Is that possible now?

> Maybe we add a grep lint in `pre-commit` if so; otherwise folks are going to forget?

I did not know of a solution that would make that possible. This is a good method at this stage!

Reply:
`prql_no_test` will skip over creating a comparison table: https://github.com/prql/prql/blob/13d0060dfe80ee3ab098b85207ef6fae9d367a55/web/book/src/lib.rs#L155

(Though do we sometimes _want_ a comparison table?)

Reply:
I think the issue here is the gap between rendering on GitHub and rendering on Book.
That is, if `prql` code blocks are recognized on GitHub in the future (#1636), Changelog will need to use `prql` code blocks, but may not want to output `sql` on the Book.

Reply:
Considering that the syntax has changed and past `prql` code block can no longer be compiled, Changelog still needs to disable `prql` code evaluation on a file-by-file basis.

Reply:
> Considering that the syntax has changed and past `prql` code block can no longer be compiled, Changelog still needs to disable `prql` code evaluation on a file-by-file basis.

We could use a comment such as `prql-book: ignore-file` here and react to that by ignoring the rest of the file: https://github.com/prql/prql/blob/13d0060dfe80ee3ab098b85207ef6fae9d367a55/web/book/src/lib.rs#L111

(I would much rather we did that than manually exclude `CHANGELOG.md`, even though it's more work)

Reply:
> That is, if `prql` code blocks are recognized on GitHub in the future (#1636), Changelog will need to use `prql` code blocks, but may not want to output `sql` on the Book.

Yeah, we could use something like 

`````
```prql,no-test
derive x = 5
```
`````

...which GitHub would still handle as PRQL (at least it seems to work with `sql` just now)

But yes, we would have to go back and change old language blocks when the syntax no longer works

Reply:
I remember the docusaurus syntax.

We can add information separated by spaces.
https://docusaurus.io/docs/markdown-features/code-blocks

````md
```jsx title="/src/components/HelloCodeTitle.js"
function HelloCodeTitle(props) {
  return <h1>Hello, {props.name}</h1>;
}
```
````

...is rendered as

```jsx title="/src/components/HelloCodeTitle.js"
function HelloCodeTitle(props) {
  return <h1>Hello, {props.name}</h1>;
}
```

In any case, I am sure we can discuss these new features later.

Reply:
Yes, I actually thought I'd make a change to use a list, quick PR at https://github.com/PRQL/prql/pull/2350/files

(doesn't affect this PR though)

Reply:
Wow, wonderfully fast work, thanks!
So can we merge this PR now?

Reply:
Yes!

Reply:
> What should the instructions for including change blocks here be? "Always use `elm`"? "Only use `prql` if you want a comparison table"?

Good point.

It seems we need to stay with `elm` for now.
For now, it seems that when the `prql` code block is evaluated, a header `PRQL` is added and the display is corrupted.

![image](https://user-images.githubusercontent.com/50911393/229096928-8664de81-971e-4fb8-a0d6-a0525db574da.png)

https://github.com/PRQL/prql/blob/fe1b7d487f300d51b72723ad9d3fdac9a84238dc/CHANGELOG.md?plain=1#L315-L327